fault/tester/staged_tester.py

```diff
@@ -393,7 +396,16 @@ def __str__(self):
         return self.name
 
 
-class LoopTester(Tester):
+class NoClockInit:
```

Comment:
Seems like it would be better to have `TesterBase` class have a default implementation of `def init_clock(self): pass`. And potentially have a `SynchronousTester` which provides the logic above by default.

Reply:
That would make sense, but there's an issue with backwards compatibility if we remove this init clock logic from the base tester

Reply:
(i.e. existing TBs that rely on the default clock initialization logic that don't use synchronous tester)

Reply:
I think I see the problem. Things like `LoopTester` need to inherit from `Tester` and not `TesterBase` to get all the implementations of normal functions. So we have to override `init_clock` explicitly somehow in `LoopTester` etc.

This seems fine to me in light of that.